README.md

```diff
@@ -218,16 +218,65 @@ class Config : public Owned<::z_owned_config_t> {
 
 Classes which protects corresponding so-called `owned` structures from copying, allowing move semantic only. Corresponding utility functions like `z_check`, `z_null`, `z_drop` are integrated into the `Owned` base template.
 
-* Closures
+#### Closures
 
-It's classes representing callback structures:
+It's classes representing [zenoh-c] and [zenoh-pico] callback structures:
 ```C++
 typedef ClosureMoveParam<::z_owned_closure_reply_t, ::z_owned_reply_t, Reply> ClosureReply;
 typedef ClosureConstPtrParam<::z_owned_closure_query_t, ::z_query_t, Query> ClosureQuery;
 ```
 
 They allows to wrap C++ invocable objects (fuctions, lambdas, classes with operator() overloaded) and pass them as callbacks to zenoh.
 
+The `ClosureConstPtrParam` types accepts objects, invocable with `const Foo*` parameter. For example
+```C++
+session.declare_subscriber("foo/bar", [](const Sample *sample) { ... });
+```
+or
+```C++
+void on_query(const Query* query) { ... };
+...
+auto queryable = std::get<Queryable>(session.declare_queryable("foo/bar", on_query);
+```
+
+The `ClosureMoveParam` types accepts types, invocable with `Foo`, `Foo&` or `Foo&&`. Callback may take ownership of the reference parameter passed 
+or do nothing and leave to caller to drop it.
+```C++
+session.get("foo/bar", "", [](Reply reply) { ... });
+session.get("foo/bar", "", [](Reply& reply) { ... });
+session.get("foo/bar", "", [](Reply&& reply) { ... });
+```
+
+The `nullptr` or invalid value (`value.check() == false`) are passed to callback before the callback is dropped. 
+This may be changed in nearest future (see below).
+
+### Known API issues and restrictions:
+
+- Objects created by `Session` keeps reference to the stack instance of `Session` object. This means that C++ move semantics
+doesn't supported correctly. See the code below (which is copied with a few changes from [here](examples/simple/universal/z_simple.cxx)). 
+**This is planned to be fixed in next release, on [zenoh-c] or [zenoh] level**
```

Comment:
@milyin @Mallets Can you add a link to an open issue that we could subscribe to?